lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
@afchung , I tried to do heap compaction . Problem is GC.Setting does not
have the variable/option you mentioned. I read about it and it says option
is there 4.5.1 onwards. My visual studio shows the version to be 4.6. So I
am confused that why I am not able to access the property.

dhruv


On Tue, Sep 29, 2015 at 9:42 AM, Andrew Chung <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40695776>:
>
> >              {
> > +                GC.Collect();
>
> @dkm2110 <https://github.com/dkm2110> Given that batch-processing is the
> primary use case for IMRU, providing an option for GC.Collect() seems to
> be reasonable here. So long as you're already invoking garbage collection,
> do you think it's also worth it to investigate how long it takes for a
> Large-Object-Heap compaction? @markusweimer
> <https://github.com/markusweimer> is right that currently HDInsight
> container memory limits are not enforced by the CLR, and we've also had to
> deal with that by periodically calling GC.Collect() in ASA after
> calculating Process.GetCurrentProcess().PrivateMemorySize64,
> unfortunately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40695776>.
>